wled00/cfg.cpp

```diff
@@ -57,14 +57,22 @@ bool deserializeConfig(JsonObject doc, bool fromFS) {
 #endif
 
   JsonObject id = doc["id"];
-  getStringFromJson(cmDNS, id[F("mdns")], 33);
-  getStringFromJson(serverDescription, id[F("name")], 33);
+  // legacy behaviour
+  getStringFromJson(hostName, id[F("mdns")], sizeof(hostName));
+  if (strlen(hostName) == 0) {
+    mDNSenabled = false; // if no host name is set, disable mDNS
+    sprintf_P(hostName, PSTR("wled-%.*s"), 6, escapedMac.c_str() + 6);
```

Comment:
This needs to be changed to use the existing logic, for example my current hostname of window is because i have the WLED device called Window

Reply:
This uses existing logic. If `{"id":{"mdns":....}, ...}` exists, it uses that name. Only if someone disabled mDNS (by entering empty name) then new hostname is generated.

Reply:
the non-mDNS hostname is wled-{serverDescription}, so actually you get "wled-window" not just "window"

wled-window.ftriz.box for example - when name in server description in the UI settings is "Window"